### PR TITLE
Plotdata

### PR DESCRIPTION
**Pull request by enchom**
Contains parts of fmcw_plot.m and SidiElAidiApp.m - the parts that prepare data produced by fmcw_range for plotting. Contains also a slight package refactoring.

### CODE REVIEW

File: computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/plotdata/PlotDataGenerator2D.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        this.win = win;
+
+        computePlotData();
+    }
+
+    /**
+     * Creates a generator using the parameters. Uses default padding and window function. The
+     * parameters correspond to values used in fmcw_range.
+     */
+    public PlotDataGenerator2D(Burst burst, double maxrange) {
```
Comment (patricoferris) [inline computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/plotdata/PlotDataGenerator2D.java]: Maybe pull the constructors to the top of the file - just to be consistent with most of our current Java files. Not super important though.

Comment (enchom) [reply computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/plotdata/PlotDataGenerator2D.java]: Will do. I am used to ordering functions based on dependencies between them, but constructors on top makes a lot of sense.

Comment (enchom) [reply computationEngine/src/main/java/uk/ac/cam/cl/juliet/computationengine/plotdata/PlotDataGenerator2D.java]: Will do. I am used to ordering functions based on dependencies between them, but constructors on top makes a lot of sense.

**Review by patricoferris — APPROVED** [review]
Tbh everything looks good to me - I can't speak about the underlying implementations of the algorithms but I trust these are correct - as for the interface for me this works 👍 

For graphing with MPAndroidCharts I will have to convert to `floats` do you think this will cause too much of an issue? 

If you could also just pull the constructors to the top that would be consistent with the rest of the code but tbh I don't think this matters too much.

**Pull request merged**